pydantic_settings/sources.py

```diff
@@ -144,6 +144,63 @@ def _extract_field_info(self, field: FieldInfo, field_name: str) -> List[Tuple[s
 
         return field_info
 
+    def _replace_field_names_case_insensitively(self, field: FieldInfo, field_values: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Replace field names in values dict by looking in models fields insensitively.
+
+        By having the following models:
+
+            class SubSubSub(BaseModel):
```

Comment:
one comment (doesn't matter much here since I guess we won't add this to the documentation) - if you want code in docsting, please use markdown with code fences so they examples render properly.

Reply:
Nice one. Done!